src/cljs/main/broadfcui/common/table/body.cljs

```diff
@@ -0,0 +1,122 @@
+(ns broadfcui.common.table.body
+  (:require
+    [dmohs.react :as react]
+    [broadfcui.common :as common]
+    [broadfcui.common.flex-utils :as flex]
+    [broadfcui.common.icons :as icons]
+    [broadfcui.common.table.utils :as table-utils]
+    [broadfcui.utils :as utils]
+    ))
+
+
+(defn- flex-params [width]
+  (if (= width :auto)
+    {:flexBasis "auto" :flexGrow 1 :flexShrink 1}
+    {:flexBasis width :flexGrow 0 :flexShrink 0}))
+
+
+(defn- header [{:keys [joined-columns sort-column sort-order set-sort style
+                       start-column-drag column-reset allow-no-sort?]}]
+  [:div {:style (merge {:display "flex"} (:row style) (:header-row style))}
+   (map-indexed
+    (fn [index {:keys [id width initial-width header visible? resizable? sortable?]}]
+      (when visible?
+        [:div {:style (merge (flex-params width)
+                             {:position "relative" :cursor (when sortable? "pointer")}
+                             (when resizable? (or (:resize-tab style)
+                                                  {:borderRight "1px solid" :marginRight -1})))}
+         [:div {:style (merge {:display "flex" :width width}
+                              (:cell style)
+                              (:header-cell style))
+                :onClick (when sortable?
+                           #(cond (not= sort-column id) (set-sort id :asc)
+                                  (= sort-order :asc) (set-sort id :desc)
+                                  allow-no-sort? (set-sort nil nil)
+                                  :else (set-sort id :asc)))}
+          [:div {:style {:flex "1 1 auto" :overflow "hidden" :textOverflow "ellipsis"}}
+           header]
+          flex/spring
+          (when (= id sort-column)
+            (icons/icon {:style {:margin "0 0.4rem 0 0.1rem"}}
+                        (if (= :asc sort-order) :sort-asc :sort-desc)))]
+         (when resizable?
+           [:div {:style {:position "absolute" :cursor "col-resize"
+                          :right -11 :top 0 :width 21 :height "100%" :zIndex 1}
```

Comment:
Is there some way to canonize this behind a named variable?
(Also, any reason to keep `:top` here?)